modules/ROOT/pages/migration-from-5x.adoc

```diff
@@ -3,88 +3,585 @@
+
+The `primary` property was boolean, allowing for only two states: `true` and `false`.
+
+The new `buttonType` property can take any of three states: `'primary'`, `'secondary'`, and `'toolbar'`.
+
+Setting `buttonType: 'primary'` is the same as setting `primary: true`.
+
+Setting `buttonType: 'secondary'` is the same as setting `primary: false`.
+
+=== `change` event
+
+Previously, the `change` event was triggered by the first modification made to a {productname} editor instance. However, subsequent changes did not cause the event to fire until focus was switched away from the editor.
+
+For example, typing a single character in to the {productname} editor text-entry area triggered the `change` event. Consequent typing, or selecting of the entered character and modifying it by setting it to *bold* or _italic_, did not trigger further `change` events. Making further modifications and then switching focus away from the editor did trigger this event, however.
+
+As of {productname} 6, the `change` event is not fired unless and until
+
+. focus is switched away from the editor; and
+. changes have been made in the editor since focus was switched to it.
+
+NOTE: It is still possible to listen for modification to a {productname} editor instance before focus is changed. To do this, listen for the `dirty` event, rather than the `change` event.
+
+=== `disabled` properties renamed to `.enabled` in {productname} UI classes
+
+Previously, {productname} UI classes provided a `.disabled` property.
+
+In {productname} 6, this property has been renamed to `.enabled`.
+
+Configurations that set `.disabled` must now set `.enabled`.
+
+Also, the Dialog API, which previously offered the `disable(<name>)` and `enable(<name>)` functions, now offers a single `setEnabled(<name>, <state>)` function.
+
+[[ui-elements-and-components-editor-text-entry-height]]
+=== Editor text entry height
+
+In {productname} 5.x, editors without a `height` value specified in the {productname} configuration defaulted to a 200px high text entry area. In {productname} 6.0, editors without a `height` value specified in the {productname} configuration default to a 400px high text entry area.
+
+As was the case in {productname} 5.x, this new default only applies to editors in `iframe` mode, not `inline` mode.
+
+The new default can be over-ridden by explicitly setting a `height` value in the {productname} configuration. So setting this to `200px` will recreate the previous default presentation.
+
+Also, setting the element’s CSS `height` to a value larger than 400px will increase the height of the text entry area. Setting the elmement’s CSS `height` to a value less than 400px will *not* reduce the height of the text entry area, however.
+
+NOTE: While {productname} 6.0 honours the `height` setting in {productname} configuration, to prevent text entry areas from disappearing during re-sizing, the `min-height` value (100px) cannot be over-ridden.
```

Comment:
```suggestion
NOTE: While {productname} 6.0 honours the `height` setting in {productname} configuration, to prevent the editor UI from disappearing during re-sizing, the `min-height` value (100px) cannot be over-ridden.
```

Reply:
```suggestion
NOTE: While {productname} 6.0 honours the `height` setting in {productname} configuration, to prevent the editor UI from disappearing during re-sizing, the `min-height` value (100px) cannot be over-ridden. This is to prevent an unrecoverable bug that prevented users being able to interact with the editor.
```
Or this, to give a bit more reasoning?